rfc/halyard-lite.md

```diff
@@ -0,0 +1,473 @@
+Kubernetes. This document proposes replacing Halyard with a smaller tool that
+integrates better into the Kubernetes ecosystem. Throughout this document, the new
+tool is referred to by the temporary name _Halyard Lite_.
+
+### Goals and Non-Goals
+
+Goals:
+* Improve the install process for users deploying to Kubernetes by providing an
+install path that is Kubernetes-native and integrates well with the Kubernetes
+ecosystem
+* Reduce the maintenance burden of Halyard, in particular by reducing the toil
+required to support new configuration parameters
+
+Non-goals:
+* This document focuses primarily on the motivation behind replacing Halyard with
+a smaller tool as well as the broad design and surface of of this tool. A more detailed
+technical design of this tool is not in scope for this document.
+
+## Motivation and Rationale
+
+The primary purpose of Halyard is to simplify the process of configuring, updating,
+and operating Spinnaker.
+
+Without Halyard, installing Spinnaker involves:
+*   Determining the version of each microservice you’d like to deploy.  In general,
+this would involve picking a top-level Spinnaker version (ex: 1.16.1) and looking
+up the version of each microservice tied to that top-level version.
+*   Writing a `service.yml` config file for each microservice containing the
+configuration you want for your Spinnaker installation, and copy it to the
+correct directory.
+    *   The configuration parameters accepted by each microservice are in general
+    not well documented, so writing these config files would require a fairly
+    detailed understanding of each microservice.
+
+Installing to a Kubernetes cluster has the additional complexity of:
+*   Adding each service’s config file as a `ConfigMap` to your cluster
+*   Writing a workload (`Deployment`/`ReplicaSet`) manifest for each microservice,
+making sure to mount the appropriate configuration into each workload.
+*   Writing a `Service` for any workloads (deck, gate) that should be exposed
+outside of the cluster.
+
+With Halyard, this process is greatly simplified.  In particular, a user can
+install Spinnaker to Kubernetes via:
+*   Configuring Halyard with the top-level Spinnaker version to install, without
+needing to know about the individual microservice versions
+*   Adding all configuration parameters to a single config file, either by editing
+the YAML directly or by running `hal config` commands
+*   Running `hal deploy apply` to deploy Spinnaker
+
+While Halyard has greatly simplified the install process, it has a number of shortcomings.
+
+### Not Kubernetes Native
+
+As Spinnaker positions itself as a Kubernetes-native deployment tool, operators
+are expecting to deploy and operate Spinnaker using the same Kubernetes-native
+tools they use for other software.  In particular, the workflow for updating
+Spinnaker using Halyard is to SSH to a VM (or a pod) and run a series of imperative
+commands.
```

Comment:
Adding a bit more color - the current implementation doesn't lend itself well to any style of GitOps or IAC (Infrastructure as Code) workflows because of Halyard's client/server nature. While are have seen a number of users work around this, the process for doing so is clumsy and error prone making it difficult to automate.

Reply:
Well said, I've added this directly to the RFC as it provides useful context.